browser/ui/brave_actions/brave_action_icon_with_badge_image_source.cc

```diff
@@ -19,6 +20,52 @@
 #include "ui/gfx/image/image_skia_operations.h"
 #include "ui/gfx/skia_paint_util.h"
 
+using namespace brave_actions;
+
+void BraveActionIconWithBadgeImageSource::Draw(gfx::Canvas* canvas) {
```

Comment:
I don't think we want to copy all this code from `IconWithBadgeImageSource::Draw`
That's really the only thing worse than patching. What exactly are we trying to do here?

Reply:
maybe we can patch in a virtual method and do something like
```
CustomScaleImageSkiaRep(....) ||
skia.AddRepresentation(ScaleImageSkiaRep(...
``` where the default implementation of `CustomScaleImageSkiaRep` just returns false?

Reply:
I'll do something like this. However, it's not just the retrieving of the image, but also the coords of painting due to different alignment requirements.